### PR TITLE
[Node] support openai SDK version ^4.24.1 embeddings api

### PR DESCRIPTION
**Pull request by koolamusic**
**Changes:**
- Updated `OpenAIEmbeddingFunction` to align with OpenAI SDK v4.24.
- Removed the configuration parameter from the OpenAI client initialization for simplicity.
- Implemented Sinon stubs in tests to mock `openai.embeddings.create` method, eliminating the need for a real API key.

**Impact:**
- Streamlines OpenAI client usage in line with the latest SDK.

### CODE REVIEW

**Comment by changhiskhan** [comment]
hey there are you able to just run `npm run lint -- --fix` to fix the lint issues?

File: node/package.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
     "eslint-plugin-n": "^15.7.0",
     "eslint-plugin-promise": "^6.1.1",
     "mocha": "^10.2.0",
-    "openai": "^3.2.1",
+    "openai": "^4.24.1",
```
Comment (changhiskhan) [inline node/package.json:49]: could go to 4.24.3 now

**Review by changhiskhan — COMMENTED** [review]
looks good in general. Thanks so much!
Should be good to go once you resolve the merge conflicts.

excited to get this update in.

File: node/package.json (hunk 1 above)
Comment (koolamusic) [reply node/package.json:49]: @changhiskhan done

File: node/src/embedding/openai.ts [hunk 2]
```diff
@@ -13,36 +13,44 @@
...
+    const configuration = {
       apiKey: openAIKey
-    })
-    this._openai = new openai.OpenAIApi(configuration)
+    }
+
+    this._openai = new Openai(configuration)
     this._modelName = modelName
+
+    console.log({ op: this._openai.embeddings.create, md: this._modelName })
```
Comment (changhiskhan) [inline node/src/embedding/openai.ts:42]:
```suggestion
```

Remove this console log

**Review by changhiskhan — APPROVED** [review]
looks good other than the console log. thanks!

**Comment by changhiskhan** [comment]
ok i fixed the merge conflicts. and merged this manually. thanks for the contrib

**Comment by koolamusic** [comment]
My pleasure.

**Pull request merged**